tests/typecheck/managers/test_managers.yml

```diff
@@ -169,28 +169,29 @@
     main: |
         from myapp.models import AbstractPerson, Book
         reveal_type(AbstractPerson.abstract_persons)  # N: Revealed type is "django.db.models.manager.Manager[myapp.models.AbstractPerson]"
-        reveal_type(Book.published_objects)  # N: Revealed type is "myapp.models.PublishedBookManager[myapp.models.Book]"
+        reveal_type(Book.published_objects)  # N: Revealed type is "myapp.models.PublishedBookManager"
         Book.published_objects.create(title='hello')
-        reveal_type(Book.annotated_objects)  # N: Revealed type is "myapp.models.AnnotatedBookManager[myapp.models.Book]"
+        reveal_type(Book.annotated_objects)  # N: Revealed type is "myapp.models.AnnotatedBookManager"
         Book.annotated_objects.create(title='hello')
     installed_apps:
         - myapp
     files:
         -   path: myapp/__init__.py
         -   path: myapp/models.py
             content: |
+                from typing import ClassVar
                 from django.db import models
 
                 class AbstractPerson(models.Model):
-                    abstract_persons = models.Manager['AbstractPerson']()
+                    abstract_persons: ClassVar[models.Manager["AbstractPerson"]] = models.Manager['AbstractPerson']()
```

Comment:
Why do we need this change?

Reply:
Looked around a bit and It comes from the plugin no longer forcing `is_classvar = True` on an all manager `Var`s

I don't think we should need to set `ClassVar` explicitly either, but mypy otherwise gives us an error on attribute access

```
main:2: error: Access to generic instance variables via class is ambiguous  [misc]
```

Not a big thing that the plugin forces `is_classvar = True`, but optimally I don't think it should be necessary. Not sure what to make of the error though, I don't see why it's a generic instance variable..

Reply:
Right, I think I've found the problem. Found that one exception that avoids the error accessing generic instance variable via class is "Self type wrapped in ClassVar"

Ref: https://github.com/python/mypy/blob/570b90a7a368f04c64f60af339d0ac1808c49c15/mypy/checkmember.py#L1065-L1070

But our plugin blindly adds `Self` as a generic argument to a manager, regardless of `is_classvar=True` or not.

https://github.com/typeddjango/django-stubs/blob/972a4e529b733820c2d9317310b279c6823b3f41/mypy_django_plugin/transformers/querysets.py#L42-L54

Which means that the error will appear for e.g. `Manager[Self]`

Reply:
I suppose there's 2 ways out to keep working with `Self`:

1. Force `is_classvar = True`
2. Don't add a type argument `Self` when `is_classvar = False`

Reply:
I went with alternative 1. and we're forcing ClassVar, as that's what we did previously. That should leave those parts unchanged I'm thinking.

Reply:
Well, now I just found that I went down this exact rabbit hole before: https://github.com/typeddjango/django-stubs/pull/1649#discussion_r1313778655

Had completely forgotten about that..